kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/CrunchDatasets.java

```diff
@@ -231,6 +232,38 @@ public static Target asTarget(URI uri) {
     }
   }
 
+  public static <E, K> PCollection<E> partitionAndSort(PCollection<E> collection,
+      View<E> view, MapFn<E, GenericData.Record> getSortKeyFn, Schema sortKeySchema) {
+    return partitionAndSort(collection, view, getSortKeyFn, sortKeySchema, -1);
+  }
+
+  public static <E, K> PCollection<E> partitionAndSort(PCollection<E> collection,
+      View<E> view, MapFn<E, GenericData.Record> getSortKeyFn, Schema sortKeySchema,
+      int numWriters) {
+    return partitionAndSort(collection, view, getSortKeyFn, sortKeySchema, numWriters, 1);
+  }
+
+  public static <E, K> PCollection<E> partitionAndSort(PCollection<E> collection,
+      View<E> view, MapFn<E, GenericData.Record> getSortKeyFn, Schema sortKeySchema,
```

Comment:
I agree it would be better to make this more flexible. I'll have a look at doing that.